docs/DeveloperGuide.md

```diff
@@ -6,8 +6,77 @@
+`editExercise()` function. This feature utilizes the `setPerformance()`and `getReps()` methods from the 
+`ExerciseStation` classes to edit the repetitions and timings for the user’s selected 
+exercises. Additionally, it calculates the points the user will earn for each exercise based on the updated values.
+
+#### Step 1: Logging a New Training Session
+
+When the user logs a new training session, an instance of the `TrainingSession` class is created. This instance 
+initializes an EnumMap, which populates the various `ExerciseStation` instances with their initial values. 
+Below is a representation of how the `ExerciseStations` are initialized:
+
+![TrainingSessionInitialState.png](TrainingSessionInitialState.png)  
+
+#### Step 2: Editing a Training Session
+
+When the user wishes to edit a training session, they specify an `Exercise` Enum, and the reps/timing to be inputted,
+which is passed to the`editExercise` function. This function calls the relevant methods to update the repetitions or 
+timings and calculates the corresponding points for the specified exercise.
+
+The following sequence diagram illustrates the function calls involved in this process:
+
+![editExerciseSequenceDiagram.png](editExerciseSequenceDiagram.png)
+
+Additionally, the state diagram below shows the end state of the `editExercise` function after execution of the command,
+`editExercise(Exercise.PULL_UP, 1)`:
+
+![TrainingSessionEditState.png](TrainingSessionEditState.png)
+
+### Points Calculation Feature
+
+The **points calculation feature** is a significant part of the `ExerciseStation` system. It allows for the calculation
+of user-specific points based on their performance in various exercises (e.g., pull-ups, sit-ups). This process 
+involves interaction between the `ExerciseStation` and the `Calculator` classes, ensuring that the correct points are
+assigned based on predefined lookup tables.
+
+#### 1. Class Interaction Overview
+
+Each `ExerciseStation` subclass (e.g., `PullUpStation`, `SitUpStation`) has its own implementation of the `getPoints()` 
+method. The main responsibility of this method is to invoke the `calculatePoints()` function from the 
+respective **calculator** class (e.g., `PullUpCalculator`, `SitUpCalculator`), which holds the points calculation logic.
+
+![getPointsSequenceDiagram.png](getPointsSequenceDiagram.png)
+##### Sequence of Events:
+
+1. **User Inputs Performance**: The user’s performance (e.g., number of pull-ups) is passed to the
+   `setPerformance()` method in the exercise station.
+
+2. **Invoke Points Calculation**: Once the performance is set, the `getPoints()` method is called. 
+   This method interacts with the calculator class (e.g., `PullUpCalculator`) to compute the points.
+
+3. **Calculator Logic**: The calculator class uses a lookup table, which maps the user's performance 
+   to points based on their age and gender. The points are returned to the exercise station, where they are stored.
+
```

Comment:
Are the user stories below still unchanged from the template?
Where are the user stories for your project?